Mathlib/Data/List/Basic.lean

```diff
@@ -2983,6 +2983,33 @@ theorem filter_false (l : List α) :
     filter (fun _ => false) l = [] := by induction l <;> simp [*, filter]
 #align list.filter_false List.filter_false
 
+theorem filter_eval_true (l : List α)
+    (h : ∀ a ∈ l, p a = true) : List.filter p l = l := by
```

Comment:
This is subsumed by [`List.filter_eq_self`](https://leanprover-community.github.io/mathlib4_docs/Batteries/Data/List/Lemmas.html#List.filter_eq_self)